Orange/widgets/unsupervised/owsom.py

```diff
@@ -0,0 +1,858 @@
+        self.grid_cells = np.full((self.size_y, self.size_x), None)
+        for y in range(self.size_y):
+            for x in range(self.size_x - (y % 2) * self.hexagonal):
+                if self.hexagonal:
+                    cell = QGraphicsPathItem(_hexagon_path)
+                    cell.setPos(x + (y % 2) / 2, y * sqrt3_2)
+                else:
+                    cell = QGraphicsRectItem(x - 0.5, y - 0.5, 1, 1)
+                self.grid_cells[y, x] = cell
+                cell.setPen(self._grid_pen)
+                self.grid.addToGroup(cell)
+        self.scene.addItem(self.grid)
+
+    def get_member_indices(self, x, y):
+        i, j = self.cells[x, y]
+        return self.member_data[i:j]
+
+    def _recompute_som(self):
+        if self.cont_x is None:
+            return
+
+        class Optimizer(QObject):
+            update = Signal(float, np.ndarray, np.ndarray)
+            done = Signal(SOM)
+            stopped = Signal()
+
+            def __init__(self, data, widget):
+                super().__init__()
+                self.som = SOM(
+                    widget.size_x, widget.size_y,
+                    hexagonal=widget.hexagonal,
+                    pca_init=widget.initialization == 0,
+                    random_seed=0 if widget.initialization == 2 else None)
+                self.data = data
+                self.widget = widget
+
+            def callback(self, progress):
+                self.update.emit(
+                    progress,
+                    self.som.weights.copy(), self.som.ssum_weights.copy())
+                return not self.widget.stop_optimization
+
+            def run(self):
+                try:
+                    self.som.fit(self.data, N_ITERATIONS,
+                                 callback=self.callback)
+                    # Report an exception, but still remove the thread
+                finally:
+                    self.done.emit(self.som)
+                    self.stopped.emit()
+
+        def update(_progress, weights, ssum_weights):
+            progressbar.advance()
+            self._assign_instances(weights, ssum_weights)
+            self._redraw()
+
+        def done(som):
+            self.enable_controls(True)
+            progressbar.finish()
+            self._assign_instances(som.weights, som.ssum_weights)
+            self._redraw()
+            # This is the first time we know what was selected (assuming that
+            # initialization is not set to random)
+            if self.__pending_selection is not None:
+                self.on_selection_change(self.__pending_selection)
+                self.__pending_selection = None
+            self.update_output()
+
+        def thread_finished():
+            self._optimizer = None
+            self._optimizer_thread = None
+
+        progressbar = gui.ProgressBar(self, N_ITERATIONS)
+
+        self._optimizer = Optimizer(self.cont_x, self)
+        self._optimizer_thread = QThread()
```

Comment:
Add a `self._optimizer_thread.setStackSize(5 * 2 ** 20)`.
By default `QThread` leaves the stack size to the platform default, which is insufficient for [openblas that comes with numpy's wheel distribution](https://github.com/numpy/numpy/issues/11551) (5MB is used by [python scheduled threads](https://github.com/python/cpython/blob/d0eeb93/Python/thread_pthread.h#L41)).

Reply:
Done. Thanks.